app/controllers/users_controller.rb

```diff
@@ -1,16 +1,25 @@
 class UsersController < ApplicationController
+  include SocialAccountsHelper
   before_action :authorize, except: %i[show index]
+  helper_method :generate_link,  :get_supported_social_networks
+
 
   def show
     @user = User.find(params[:id])
   end
 
-  def edit; end
+  def edit
+    @user = User.find(params[:id])
+    # prototype for create social account form
+    @social_account = @user.social_accounts.build #
+  end
 
   def update
-    return render :edit unless @user.update(user_params)
-
-    redirect_to @user
+    # prototype for create social account form
+    return redirect_to @user if @user.update(user_params)
+    handle_error
```

Comment:
why not adding normal flash messages here? (That would be displayed by the logic in the layout)

Reply:
My assumption was to display a heading and then all error messages as bullet points. 

<img width="1137" alt="Screenshot 2021-01-15 at 10 53 06" src="https://user-images.githubusercontent.com/32880898/104710014-de4b1f80-571f-11eb-9520-298322e1a7c5.png">

That's why the flash message partial expects a dictionary in form of {heading: "Some Heading" , messages: [messageOne, ...] } . Do you think that it makes sense to display a heading?

Reply:
Ah, I now understand what you are doing, the screenshot helped a lot!  
I've always oriented my error explanation at the [rails guide](https://guides.rubyonrails.org/active_record_validations.html#displaying-validation-errors-in-views).  
Your solution adds a little bit overhead because every controller that is using this needs to implement `handle_errors`. But fine for me. Thank you for clarification!